admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/UserDetailsServiceTest.java

```diff
@@ -27,672 +61,942 @@
 		mockRestServiceServer.expect(requestTo(userDetailsUri.toString() + "/admin?roleName=AUTH&pageStart=0&pageFetch=100"))
 		.andRespond(withSuccess().body(response).contentType(MediaType.APPLICATION_JSON));
 		userDetailsService.getUsers("AUTH",0,0,"","","");
 	}
-}
\ No newline at end of file
+	@Test (expected = DataNotFoundException.class)
+	public void getUsersTest001() {
+		Page<UserDetails> pageData=new Page<UserDetails>() {
+			@Override
+			public Iterator<UserDetails> iterator() {
+				return null;
+			}
+
+			@Override
+			public int getTotalPages() {
+				return 0;
+			}
+
+			@Override
+			public long getTotalElements() {
+				return 0;
+			}
+
+			@Override
+			public int getNumber() {
+				return 0;
+			}
+
+			@Override
+			public int getSize() {
+				return 0;
+			}
+
+			@Override
+			public int getNumberOfElements() {
+				return 0;
+			}
+
+			@Override
+			public List<UserDetails> getContent() {
+				return null;
+			}
+
+			@Override
+			public boolean hasContent() {
+				return false;
+			}
+
+			@Override
+			public Sort getSort() {
+				return null;
+			}
+
+			@Override
+			public boolean isFirst() {
+				return false;
+			}
+
+			@Override
+			public boolean isLast() {
+				return false;
+			}
+
+			@Override
+			public boolean hasNext() {
+				return false;
+			}
+
+			@Override
+			public boolean hasPrevious() {
+				return false;
+			}
+
+			@Override
+			public Pageable nextPageable() {
+				return null;
+			}
+
+			@Override
+			public Pageable previousPageable() {
+				return null;
+			}
+
+			@Override
+			public <U> Page<U> map(Function<? super UserDetails, ? extends U> converter) {
+				return null;
+			}
+
+
+		};
+		Mockito.when(userDetailsRepository.findAllByIsDeletedFalseorIsDeletedIsNull(Mockito.any(Pageable.class)))
+				.thenReturn(pageData);
+		PageDto<UserDetailsExtnDto> pageDetails = userDetailsService.getUsers(0,1,"name","DESC");
+
+	}
+
+
+	@Test (expected = DataNotFoundException.class)
+	public void getUsersByRegistrationCenterTest001(){
+
+		List<UserDetails> pageDetails = new ArrayList<UserDetails>();
+
+		Mockito.when(userDetailsRepository.findByRegIdAndIsDeletedFalseOrIsDeletedIsNull(Mockito.any(String.class),Mockito.any()))
+				.thenReturn(pageDetails);
+		List<UserDetailsExtnDto> pageDetail = userDetailsService.getUsersByRegistrationCenter("",0,1,"name","desc");
+
+	}
+
+	@Test (expected = MasterDataServiceException.class)
+	public void getUsersByRegistrationCenterTest002(){
+
+		List<UserDetails> pageDetails = new ArrayList<UserDetails>();
+		Mockito.when(userDetailsRepository.findByRegIdAndIsDeletedFalseOrIsDeletedIsNull(Mockito.any(String.class),Mockito.any())).thenThrow(DataAccessLayerException.class);
+
+		List<UserDetailsExtnDto> pageDetail = userDetailsService.getUsersByRegistrationCenter("",0,1,"name","ASC");
+
+	}
+	@Test(expected = NullPointerException.class)
+	public void validateZoneTest001(){
+		List<Zone> subZones=new ArrayList<>();
```

Comment:
Debug this method and check why you are getting this null pointer exception and let me know

Reply:
I Debugged the method and fund out that the String value=null,ClassLoader classLoader=null and Hashtable<Object,Object> staticProp=null.